users/signals.py

```diff
@@ -0,0 +1,27 @@
+from django.db.models.signals import post_save, pre_save
+from django.dispatch import receiver
+from .models import User, Profile, UserSettings
+from .helpers import nickname_from_email
+
+
+# Do I need other signals?
+
+@receiver(post_save, sender=User)
+def create_profile(sender, instance, created, **kwargs):
+    if created:
+        profile = Profile.objects.create()
+        if instance.profile and instance.nickname:
```

Comment:
this instance.profile at this moment will always be false. So only`nickname_from_emai` which is wrong. Details below.